frontend_vue/src/utils/incidentAlertService.ts

```diff
@@ -0,0 +1,176 @@
+ * or 'HTTP' if not found.
+ */
+function hasChannelCustomLabel(type: string) {
+  return INCIDENT_CHANNEL_TYPE_LABELS.hasOwnProperty(type)
+    ? INCIDENT_CHANNEL_TYPE_LABELS[
+        type as keyof typeof INCIDENT_CHANNEL_TYPE_LABELS
+      ]
+    : 'HTTP';
+}
+
+/**
+ * Formats a given string by replacing underscores with spaces and capitalizing the first letter.
+ */
+function formatKey(string: string) {
+  const formattedString = string
+    .replace(/_/g, ' ')
+    .replace(/^(.)/, function (match, group) {
+      return group.toUpperCase();
+    });
+
+  return formattedString;
+}
+
+/**
+ * Formats the incident details object by applying custom labels and formatting keys.
+ */
+export function formatLabels(
+  incidentDetails: FormattedHitsType | keyof FormattedHitsType
+) {
+  try {
+    return Object.entries(incidentDetails).reduce(
+      (acc: Record<string, unknown>, [key, val]) => {
+        const formattedKey = INCIDENT_DETAIL_CUSTOM_LABELS.hasOwnProperty(key)
+          ? INCIDENT_DETAIL_CUSTOM_LABELS[
+              key as keyof typeof INCIDENT_DETAIL_CUSTOM_LABELS
+            ]
+          : formatKey(key);
+
+        if (typeof val === 'object' && val !== null) {
+          acc[formattedKey] = formatLabels(
+            val as unknown as keyof FormattedHitsType
+          );
+        } else {
+          acc[formattedKey] = val;
+        }
+
+        return acc;
+      },
+      {}
+    );
+  } catch (error) {
+    console.error(`Error in formatting labels: ${error}`);
+    return incidentDetails;
+  }
+}
+
+/**
+ * Formats the label of a token type based on the given token.
+ */
+export function formatTokenTypeLabel(token: string) {
+  const token_const: string | undefined = Object.values(TOKENS_TYPE).find(
+    (value) => value === token
+  );
+  return token_const ? tokenServices[token_const].label : token;
+}
+
+/**
+ * Removes keys and values from an object if the value is null, an empty object, an empty array,
+ * or if all nested keys and values are removed.
+ */
+export function removeNullEmptyObjectsAndArrays(
+  incidentDetails: FormattedHitsType
+): FormattedHitsType {
+  try {
+    return Object.entries(incidentDetails).reduce((acc, [key, value]) => {
+      if (
+        value !== null &&
+        value !== undefined &&
+        !(typeof value === 'object' && Object.keys(value).length === 0) &&
+        !(Array.isArray(value) && value.length === 0)
+      ) {
+        if (typeof value === 'object') {
+          const nestedObj = removeNullEmptyObjectsAndArrays(
+            value as FormattedHitsType
+          );
+          if (Object.keys(nestedObj).length > 0) {
+            acc[key] = nestedObj as Record<string, string>;
+          }
+        } else {
+          acc[key] = value;
+        }
+      }
+      return acc;
+    }, {} as FormattedHitsType);
+  } catch (error) {
+    console.error(`Error in remove Null, Empty Objects and Arrays: ${error}`);
+    return incidentDetails;
+  }
```

Comment:
i'm still unsure why we need to do this? can these checks not just be in the html render steps? v-`if="!notEmpty(key) && !notEmpty(value)"` kinda thing?

Reply:
yeah for sure I can do it in the template. I was just happier by dividing concerns if possible. I would like to avoid cluttering the template since we are already dealing with 3 levels of object nesting. I'm afraid adding even another condition would make it quite messy to read/understand.

Reply:
That said, if you think having the check on the template makes sense, I will move to that solution

Reply:
we can try it out. it is a little difficult to read. but maybe we don't need to worry about it

Reply:
heh i meant let's try your way but i see you've moved it to the template?